src/ripple/rpc/impl/DeliveredAmount.cpp

```diff
@@ -74,7 +74,93 @@ getDeliveredAmount(
         if (getLedgerIndex() >= 4594095 ||
             getCloseTime() > NetClock::time_point{446000000s})
         {
-            return serializedTx->getFieldAmount(sfAmount);
+            auto const meta = transactionMeta.getAsObject();
+
+            // if there are no modified fields (somehow) then nothing was
+            // delivered
+            if (!meta.isFieldPresent(sfAffectedNodes))
+                return {};
+
+            bool const hasDeliverMin =
+                serializedTx->isFieldPresent(sfDeliverMin);
+            bool const hasAmount = serializedTx->isFieldPresent(sfAmount);
```

Comment:
There's something funny with the logic here.  We know that `hasAmount` is true because we wouldn't be inside the `if` that opens on line 62 if `Amount` were absent.  So `hasAmount` will always be `true` and the `return` on line 91 can never fire.

Since a `CheckCash` transaction might not have an `Amount` field, I suspect that the check on line 62 is not quite right.  Possibly line 62 should be...
```
if (serializedTx->isFieldPresent(sfAmount) || serializedTx->isFieldPresent(sfDeliverMin)
```